src/Umbraco.Core/ObjectExtensions.cs

```diff
@@ -652,5 +644,69 @@ internal static Guid AsGuid(this object value)
 	    {
 	        return value is Guid ? (Guid) value : Guid.Empty;
 	    }
-	}
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static string NormalizeNumberDecimalSeparator(string s)
+        {
+            var normalized = System.Threading.Thread.CurrentThread.CurrentCulture.NumberFormat.NumberDecimalSeparator[0];
+            return s.ReplaceMany(NumberDecimalSeparatorsToNormalize, normalized);
+        }
+
+        // gets a converter for source, that can convert to target, or null if none exists
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static TypeConverter GetCachedSourceTypeConverter(Type source, Type target)
+        {
+            return InputTypeConverterCache.GetOrAdd(new CompositeTypeTypeKey(source, target), k =>
+            {
+                var ksource = k.Type1;

```

Comment:
Even if you ensure there's no extra allocation here there's an extra null check against the factory method for every call when using the `GetOrAdd` convenience method. Calling `TryGetValue` directly in the following manner is the more efficient code. 

``` c#
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static TypeConverter GetCachedSourceTypeConverter(Type source, Type target)
{
    var key = new CompositeTypeTypeKey(source, target);

    if (InputTypeConverterCache.TryGetValue(key, out TypeConverter typeConverter))
    {
        return typeConverter;
    }

    TypeConverter converter = TypeDescriptor.GetConverter(source);
    if (converter.CanConvertTo(target))
    {
        return InputTypeConverterCache[key] = converter;
    }

    return InputTypeConverterCache[key] = null;
}
```